File: Data Structures & Algorithms/merge-strings-alternately/submission-2.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res = string(word1.length()+word2.length(),' ');
        int i =0;
       for(int i =0,j=0,c=0;c<res.length();)
       {
            if(i<word1.length())
                res[c] = word1[i],c++,i++;
            if(j<word2.length())
                res[c] = word2[j],c++,j++;
       } 
        return res;

    }
};

File: Data Structures & Algorithms/shortest-way-to-form-string/submission-2.py
import bisect

class Solution:
    def shortestWay(self, source: str, target: str) -> int:
        dic = defaultdict(list)
        for i in range(len(source)):
            dic[source[i]].append(i)
        curIndx = 0
        res = 1
        for i in range(len(target)):
            ch = target[i]
            if ch not in dic:
                return -1
            idx = bisect.bisect_left(dic[ch], curIndx)
            if idx < len(dic[ch]):
                curIndx = dic[ch][idx] + 1
            else:
                res += 1
                curIndx = dic[ch][0] + 1
        return res